components/Login.jsx

```diff
@@ -1,31 +1,78 @@
 import React from "react";
 import Link from "next/link";
+import { useState, useEffect } from "react";
 
 const Login = () => {
+  const [formInput, setFormInput] = useState({
+    email: "",
+    password: "",
+  });
+  const [formErrors, setFormErrors] = useState({});
+  const [isSubmit, setIsSubmit] = useState(false);
+  const onChangeHandler = (e) => {
+    setFormInput(() => ({
+      ...formInput,
+      [e.target.name]: e.target.value,
+    }));
+  };
+
+  const onSubmitHandler = (e) => {
+    e.preventDefault();
+    setFormErrors(validate(formInput));
+    setIsSubmit(true);
+  };
+
+  useEffect(() => {
+    if (Object.keys(formErrors.length === 0) && isSubmit) {
+      console.log(formInput);
+    }
+  });
+
+  const validate = (values) => {
+    const errors = {};
+    const regex = /^[a-zA-Z]{3,4}[0-9]{3}@ucr\.edu$/i;
```

Comment:
nothing wrong here, just curious. how does regex work? like what is this statement saying?